addons/pos_gift_card/static/src/xml/GiftCardPopup.xml

```diff
@@ -10,115 +10,74 @@
-                                        Discard
-                                    </span>
+                                    <input t-model="state.giftCardBarcode" class="giftCardPopupInputBarcode" type="text"/>
                                 </div>
                             </div>
                         </div>
 
-                        <div class="giftCardPopupContainer"
-                             t-if="state.showBarcodeGeneration &amp;&amp; state.showUseGiftCardMenu">
-                            <div>
+                        <div class="giftCardPopupContainer" t-if="!state.showMenu and state.command == 'pay'">
+                            <div class="giftCardPopupInputContainer">
                                 Gift Card Barcode:
-                                <input t-model="state.giftCardBarcode" class="giftCardPopupInput" type="text"/>
-                            </div>
-                            <div>
-                                <span class="button confirm" t-on-click="payWithGiftCard">
-                                    Confirm
-                                </span>
-                                <span class="button cancel" t-on-click="switchBarcodeView">
-                                    Discard
-                                </span>
+                                <input t-model="state.giftCardBarcode" class="giftCardPopupInputBarcode" type="text"/>
                             </div>
                         </div>
 
-                        <div class="giftCardPopupContainer"
-                             t-if="state.showBarcodeGeneration &amp;&amp; state.showGiftCardDetails">
-                            <div>
+                        <div class="giftCardPopupContainer" t-if="!state.showMenu and state.command == 'showAmount'">
+                            <div class="giftCardPopupInputContainer" style="margin: 10px">
                                 Gift Card Barcode:
-                                <input t-model="state.giftCardBarcode" class="giftCardPopupInput" type="text"/>
+                                <input t-model="state.giftCardBarcode" class="giftCardPopupInputBarcode" type="text"/>
                             </div>
                             <div>
                                 Remaining amount of the gift card:
-                                <t t-esc="state.amountToSet"/>
-                            </div>
-                            <div>
-                                <span class="button confirm" t-on-click="ShowRemainingAmount">
-                                    Confirm
-                                </span>
-                                <span class="button cancel" t-on-click="switchBarcodeView">
-                                    Discard
-                                </span>
+                                <t t-esc="env.pos.format_currency(state.amountToSet)"/>
```

Comment:
Wouldn't it be less confusing if you set the "remaining gift card amount" in a different state variable? It's a small price to pay to be less confusing.